### PR TITLE
Remove fallback publisher logo

### LINKED ISSUE #4762: Remove fallback publisher logo
## Task Description

If the user doesn't have a publisher logo set, we currently fall back to the WordPress logo.

This might not be what a publisher desires, so we might want to consider not doing that anymore.

The format would fall back to the favicon in this case.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Remove fallback publisher logo and placeholder. 

## Relevant Technical Choices

Remove existing placeholder images and remove empty attribute. 

## Testing Instructions

---

Fixes #4762

### CODE REVIEW

File: includes/Traits/Publisher.php [hunk 1]
```diff
@@ -125,26 +125,8 @@ public function get_publisher_logo() {
...
-		}
-
-		// Try Site Icon, though it is not ideal for non-Story because it should be square.
-		$site_icon_id = get_option( 'site_icon' );
-		if ( empty( $logo_image_url ) && $site_icon_id ) {
-			$logo_image_url = $this->get_valid_publisher_image( $site_icon_id );
-		}
-
 		// Fallback to serving the WordPress logo.
 		$placeholder = $this->get_publisher_logo_placeholder();
```
Comment (swissspidy) [inline includes/Traits/Publisher.php]: The WordPress logo placeholder needs to be removed

Comment (swissspidy) [reply includes/Traits/Publisher.php]: That also means removing the `fallback-wordpress-publisher-logo.png` file

File: includes/Story_Renderer/HTML.php [hunk 2]
```diff
@@ -352,10 +352,14 @@ protected function add_publisher_logo() {
 			return;
 		}
 
-		$publisher_logo = $story_element->getAttribute( 'publisher-logo-src' );
+		$publisher_logo_src = $story_element->getAttribute( 'publisher-logo-src' );
+		if ( empty( $publisher_logo_src ) || $publisher_logo_src === $this->get_publisher_logo_placeholder() ) {
+			$story_element->removeAttribute( 'publisher-logo-src' );
```
Comment (swissspidy) [inline includes/Story_Renderer/HTML.php:357]: If there is no publisher logo, need to do `$html->removeAttribute( 'amp' );` like `add_poster_images` does below

File: includes/Traits/Publisher.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Traits/Publisher.php]: It is needed for the filter, 
https://github.com/google/web-stories-wp/blob/f2c821751fbe57f0894bc7af7e8bc68b5ba2ed82/includes/Traits/Publisher.php#L159

I also think there is a possibility that stories might be hotlinking to the image / url for the placeholder. Is it a problem if we remove it?

File: includes/Story_Renderer/HTML.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Story_Renderer/HTML.php:357]: Fixed in https://github.com/google/web-stories-wp/pull/4787/commits/c45f2726751bc701a54b7c268e9b3e66947a6a1d

File: includes/Traits/Publisher.php (hunk 1 above)
Comment (swissspidy) [reply includes/Traits/Publisher.php]: Let's remove it

File: includes/Traits/Publisher.php [hunk 3]
```diff
@@ -40,6 +40,7 @@ trait Publisher {
 	 * Uses a fallback logo to always create valid AMP in FE.
 	 *
 	 * @since 1.0.0
+	 * @deprecated 1.1.0
 	 *
 	 * @return string
 	 */
```
Comment (spacedmonkey) [inline includes/Traits/Publisher.php:45]: I tried to remove the `get_publisher_logo_placeholder` method. But I have to keep it as it is used in HTML class to remove placeholder image. With this logic, current stories with posters with the placeholder will load a 404 image See https://github.com/google/web-stories-wp/pull/4787/files#diff-9c6a425252305032ebe43f069d74d517R356

File: includes/Traits/Publisher.php [hunk 4]
```diff
@@ -125,36 +126,18 @@ public function get_publisher_logo() {
...
-		}
-
-		// Try Site Icon, though it is not ideal for non-Story because it should be square.
-		$site_icon_id = get_option( 'site_icon' );
-		if ( empty( $logo_image_url ) && $site_icon_id ) {
-			$logo_image_url = $this->get_valid_publisher_image( $site_icon_id );
-		}
-
-		// Fallback to serving the WordPress logo.
 		$placeholder = $this->get_publisher_logo_placeholder();
```
Comment (spacedmonkey) [inline includes/Traits/Publisher.php:90]: Keeping variable alive for the filter and no other reason.

File: includes/Traits/Publisher.php [hunk 5]
```diff
@@ -125,36 +126,18 @@ public function get_publisher_logo() {
...
-			$logo_image_url = $placeholder;
-		}
 
 		/**
 		 * Filters the publisher's logo.
 		 *
 		 * This should point to a square image.
 		 *
 		 * @since 1.0.0
+		 * @since 1.1.0 $placeholder Is deprecated.
```
Comment (spacedmonkey) [inline includes/Traits/Publisher.php]: Not sure if there is a better way of doing this.

File: includes/Traits/Publisher.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Traits/Publisher.php]: I have marked this as deprecated but I did not remove placeholder, as it may break others plugins / code.

**Comment by spacedmonkey** [comment]
@swissspidy Any chance of a review again?

File: includes/Story_Renderer/HTML.php [hunk 6]
```diff
@@ -352,10 +352,36 @@ protected function add_publisher_logo() {
 			return;
 		}
 
-		$publisher_logo = $story_element->getAttribute( 'publisher-logo-src' );
+		$publisher_logo_src = $story_element->getAttribute( 'publisher-logo-src' );
+		if ( empty( $publisher_logo_src ) || $publisher_logo_src === $this->get_publisher_logo_placeholder() ) {
```
Comment (swissspidy) [inline includes/Story_Renderer/HTML.php:356]: Ideally we avoid checking for the placeholder image here, but it seems like we don't store the data in a way that this is possible :-(

File: includes/Traits/Publisher.php (hunk 3 above)
Comment (swissspidy) [reply includes/Traits/Publisher.php:45]: Bummer :-(

File: includes/Traits/Publisher.php (hunk 5 above)
Comment (swissspidy) [inline includes/Traits/Publisher.php]:
```suggestion
		 * @since 1.1.0 The $placeholder parameter was deprecated.
```

**Review by swissspidy — APPROVED** [review]
approved with nits

**Comment by swissspidy** [comment]
FYI, Dashboard integration tests are also failing on `main`. Known issue as per https://xwp.slack.com/archives/CCCAUAH6F/p1601955965079000

**Pull request merged**